app/assets/javascripts/discourse/app/routes/user-private-messages.js

```diff
@@ -1,20 +1,45 @@
 import Composer from "discourse/models/composer";
+import Tag from "discourse/models/tag";
 import DiscourseRoute from "discourse/routes/discourse";
 import Draft from "discourse/models/draft";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
 
 export default DiscourseRoute.extend({
+  queryParams: {
+    tag: {
+      refreshModel: true,
+    },
+  },
+
   renderTemplate() {
     this.render("user/messages");
   },
 
   model() {
-    return this.modelFor("user");
+    const user = this.modelFor("user");
+
+    return ajax(`/tags/personal_messages/${user.username_lower}`, {
+      data: { limit: this.siteSettings.max_tags_in_filter_list },
```

Comment:
If it's a site setting maybe we don't need to pass it as a parameter? The server side route could use `SiteSetting. max_tags_in_filter_list `

Reply:
I'll like to pass the limit intentionally in the params for now. I'm not sure how that route is being used but I'll be refactoring that route soon in a separate PR.